lib/RudderOption.dart

```diff
@@ -44,34 +46,30 @@ class RudderOption {
   }
 
   RudderOption putIntegration(String type, bool enabled) {
-    if (this.integrations == null) {
+    if (integrations == null) {
       integrations = new Map();
     }
-    if (type != null && enabled != null) {
-      integrations[type] = enabled;
-    }
+    integrations[type] = enabled;
     return this;
   }
 
   RudderOption putIntegrationWithFactory(
       RudderIntegration factory, bool enabled) {
-    if (this.integrations == null) {
+    if (integrations == null) {
       integrations = new Map();
     }
-    if (factory != null && enabled != null) {
-      integrations[factory.getKey()] = enabled;
+    if (factory.getKey() != null) {
+      integrations[factory.getKey()!] = enabled;
```

Comment:
Since the `getKey()` doesn't return null. We can ignore the null check here

Reply:
<img width="799" alt="Screenshot 2021-06-03 at 12 10 51 PM" src="https://user-images.githubusercontent.com/61964755/120585685-ccff7700-c464-11eb-89c2-f08e7a73887f.png">

What do you advice to do ?

Reply:
Actually `RudderIntegration.dart` is used as a super class used for the Device mode integrations and not in the core SDK.
All the Device Mode integrations will implement `RudderIntegration.dart` and its mandatory for them to return 
the `getKey()` method in those classes to return a Non-Null value.
So I feel we can ignore this warning

Reply:
I see. Unfortunately, we can't ignore as it is an error not a warning, we have to return something or make it nullable to be able to build. So which should we go with ?

Reply:
Better we go by returning `"Integration"` from the `getKey()` in `RudderIntegration.dart`
This way we meet both the constraints.
WDYT ?

Reply:
I returned `Integration` as suggested.